x/clp/types/expected_keepers.go

```diff
@@ -33,7 +33,7 @@ type AuthKeeper interface {
 }
 
 type TokenRegistryKeeper interface {
-	GetDenom(ctx sdk.Context, denom string) tokenregistryTypes.RegistryEntry
-	IsDenomWhitelisted(ctx sdk.Context, denom string) bool
-	CheckDenomPermissions(ctx sdk.Context, denom string, permissions []tokenregistryTypes.Permission) bool
+	GetDenom(registry tokenregistryTypes.Registry, denom string) *tokenregistryTypes.RegistryEntry
```

Comment:
+1 on returning pointer, IsWhitelisted becomes less necessary in this case.

Reply:
@timlind isWhitelisted is replaced with nil check now